Documentation/boostrap-new-os.md

```diff
@@ -0,0 +1,140 @@
+#### CoreCLR
+* Unless the target OS has `/etc/os-release` file that can be used to extract the RID, the `initHostDistroRid` function in the `build.sh` file needs to be updated to be able to detect the target OS and report the proper RID by setting the `__HostDistroRid` to the RID combined with the target architecture (e.g. `mynewrid-x64`).
+* In `src/.nuget/dir.props` add a new `OfficialBuildRID` element with the new RID and target architecture. So for example, `<OfficialBuildRID Include="mynewrid-x64" />`
+#### CoreFX
+* Unless the target OS has `/etc/os-release` file that can be used to extract the RID, the `initHostDistroRid` function in the `src/Native/build-native.sh` file needs to be updated to be able to detect the target OS and report the proper RID by setting the `__HostDistroRid` to the RID combined with the target architecture (e.g. `mynewrid-x64`).
+* In `dependencies.props`, the `CoreClrPackageVersion` needs to be updated to contain the version of packages built in the coreclr repo.
+* In `external/runtime/runtime.depproj`, the content of the `Version` element under `<PackageReference Include="Microsoft.NETCore.DotNetHost">` and `<PackageReference Include="Microsoft.NETCore.DotNetHostPolicy">` needs to be updated to contain the version of packages built in the core-setup repo.
+* In `external/test-runtime/XUnit.Runtime.depproj`, add `<NoWarn>NU1603</NoWarn>` to the first `<PropertyGroup>` element.
+* In `pkg/Microsoft.NETCore.Platforms/runtime.json`, add new elements describing the new RID and its relationship under the `runtimes` key. Here is an example of what we would add for Alpine Linux 3.6:
+```json
+    "alpine": {
+        "#import": [ "linux" ]
+    },
+    "alpine-x64": {
+        "#import": [ "alpine", "linux-x64" ]
+    },
+    "alpine.3.6": {
+        "#import": [ "alpine" ]
+    },
+    "alpine.3.6-x64": {
+        "#import": [ "alpine.3.6", "alpine-x64" ]
+    },
+```
+* In `pkg/Microsoft.Private.CoreFx.NETCoreApp/netcoreapp.rids.props`, add a new `OfficialBuildRID` element with the new RID and target architecture. So for example, `<OfficialBuildRID Include="mynewrid-x64" />`
+#### Core-Setup
+* Unless the target OS has `/etc/os-release` file that can be used to extract the RID, the `init_rid_plat` function in the `src/corehost/build.sh` file needs to be updated to be able to detect the target OS and report the proper RID by setting the `__rid_plat` variable. Please note that unlike the similar case in coreclr and corefx repos, this variable is set to the RID without the target architecture (so just `mynewrid` and not e.g. `mynewrid-x64`).
+* In `dependencies.props`, update the `MicrosoftPrivateCoreFxNETCoreAppPackageVersion` element to and the `MicrosoftNETCoreRuntimeCoreCLRPackageVersion` element to.
+* In `src/pkg/projects/netcoreappRIDs.props`, add a new `OfficialBuildRID` element with the new RID and target architecture. So for example, `<OfficialBuildRID Include="mynewrid-x64" />`
```

Comment:
If you are adding a new OS that doesn't have `/etc/os-release`, you also need to update:

https://github.com/dotnet/core-setup/blob/9dff3ce12574699e6d4d31722047f775d92d05fd/src/managed/Microsoft.DotNet.PlatformAbstractions/RuntimeEnvironment.cs#L30
and
https://github.com/dotnet/core-setup/blob/9dff3ce12574699e6d4d31722047f775d92d05fd/src/corehost/common/pal.unix.cpp#L307

to account for the new RID calculation.

Reply:
`/etc/os-release` is a Linux thing. BSD and SunOS distributions don't use it.

Reply:
On such systems best to use `uname`(1) or `uname`(3).

```
$ uname -a
SunOS ***.com 5.11 joyent_20141030T081701Z i86pc i386 i86pc
```

```
$ uname -a
NetBSD rugged 8.99.1 NetBSD 8.99.1 (GENERIC) #7: Mon Jul  3 15:56:30 CEST 2017  root@chieftec:/public/netbsd-root/sys/arch/amd64/compile/GENERIC amd64
```

```
$ uname -a
FreeBSD ***.net 11.1-RELEASE FreeBSD 11.1-RELEASE #0 r321309: Fri Jul 21 02:08:28 UTC 2017     root@releng2.nyi.freebsd.org:/usr/obj/usr/src/sys/GENERIC  amd64
```

```
$ uname -a
OpenBSD obsd 6.1 GENERIC#19 i386
```

Reply:
@eerhardt thanks for spotting it, I have written this paragraph before I knew that and then forgotten to update it.